vyxal/helpers.py

```diff
@@ -355,6 +355,25 @@ def graph_distance(
         )
 
 
+def group_by_function(
+    lst: VyList, function: types.FunctionType, ctx: Context
+) -> LazyList:
+    """Group a list of elements by a function"""
+
+    @lazylist
+    def gen():
```

Comment:
There’s no need to make this a LazyList. It goes over the entire list before yielding all the values at once. If you like, I could try an infinite-list-friendly implementation later but it’s probably not going to work